content/token22-on-chain.md

```diff
@@ -0,0 +1,1483 @@
+
+```bash
+solana-install init 1.18.0
+```
+
+If you run into the following error at any point attempting to build the program, that likely means you do not have the correct version of the Solana CLI installed.
+
+```bash
+anchor build
+error: package `solana-program v1.18.0` cannot be built because it requires rustc 1.72.0 or newer, while the currently active rustc version is 1.68.0-dev
+Either upgrade to rustc 1.72.0 or newer, or use
+cargo update -p solana-program@1.18.0 --precise ver
+where `ver` is the latest version of `solana-program` supporting rustc 1.68.0-dev
+```
+
+You will also want the latest version of the Anchor CLI installed. You can follow along the steps listed here to update via avm https://www.anchor-lang.com/docs/avm or simply run:
+```bash
+avm install latest
+avm use latest
+```
+
+At the time of writing, the latest version of the Anchor CLI is `0.29.0`
+
+Now, we can check our Rust version.
+
+```bash
+rustc --version
+```
+
+At the time of writing, version `1.26.0` was used for the Rust compiler. If you would like to update, you can do so via `rustup`
+https://doc.rust-lang.org/book/ch01-01-installation.html
+
+```bash
+rustup update
+```
+
+Now, we should have all the correct versions installed.
+
+### 2. Get starter code and add dependencies
+
+Let's grab the starter branch.
+
+```bash
+git clone https://github.com/Unboxed-Software/token22-staking
+cd token22-staking
+git checkout starter
+```
+
+### 3. Update Program ID and Anchor Keypair
+
+Once in the starter branch, run `anchor keys list` to get your program ID.
+
+Copy and paste this program ID in the `Anchor.toml` file:
+
+```rust
+// in Anchor.toml
+[programs.localnet]
+token_22_staking = "<YOUR-PROGRAM-ID-HERE>"
+```
+
+And in the `programs/token-22-staking/src/lib.rs` file:
+
+```rust
+declare_id!("<YOUR-PROGRAM-ID-HERE>");
+```
+
+Lastly set your developer keypair path in `Anchor.toml`.
+
+```toml
+[provider]
+cluster = "Localnet"
+wallet = "/YOUR/PATH/HERE/id.json"
+```
+
+If you don't know what your current keypair path is you can always run the Solana cli to find out.
+
+```bash
+solana config get
+```
+
+### 4. Confirm the program builds
+
+Let's build the starter code to confirm we have everything configured correctly. If it does not build, please revisit the steps above.
+
+```bash
+anchor build
+```
+
+You can safely ignore the warnings of the build script, these will go away as we add in the necessary code.
+
+Feel free to run the provided tests to make sure the rest of the development environment is set up correctly. You'll have to install the node dependencies using `npm` or `yarn`. The tests should run, but they'll all fail until we have completed our program.
```

Comment:
Honestly not worth mentioning `yarn` or any other third-party package manager. web3.js doesn't use it anymore, spl-token doesn't use it anymore, and Anchor will drop the yarn requirement in it's next release.